tdrs-backend/tdpservice/data_files/views.py

```diff
@@ -98,16 +102,17 @@ def create(self, request, *args, **kwargs):
 
         return response
 
-    def get_s3_versioning_id(self, file_name):
+    def get_s3_versioning_id(self, file_name, prefix):
         """Get the version id of the file uploaded to S3."""
         s3 = S3Client()
         bucket_name = settings.AWS_S3_DATAFILES_BUCKET_NAME
-        versions = s3.client.list_object_versions(Bucket=bucket_name)
+        versions = s3.client.list_object_versions(Bucket=bucket_name, Prefix=prefix)
         for version in versions['Versions']:
             file_path = version['Key']
             if file_name in file_path:
                 if version['IsLatest']:
-                    return (version['VersionId'])
+                    if version['VersionId'] != 'null':
+                        return (version['VersionId'])
```

Comment:
Worth `and`-ing the conditionals b/c there are no elses? I don't think it's worth code-golfing via lambda/list comprehensions.